FRAuth/FRAuth/Error/WebAuthnError.swift

```diff
@@ -15,25 +15,31 @@ import FRCore
 ///
 /// WebAuthnError represents an error captured by FRAuth SDK for WebAuthn related operation
 ///
-public enum WebAuthnError: FRError {
-    case badData
-    case badOperation
-    case invalidState
-    case constraint
-    case cancelled
-    case timeout
-    case notAllowed
-    case unsupported
-    case unknown
-}
 
+public struct WebAuthnError: FRError {
+    public enum WebAuthnErrorCases: FRError {
+        case badData
+        case badOperation
+        case invalidState
+        case constraint
+        case cancelled
+        case timeout
+        case notAllowed
+        case unsupported
+        case unknown
+    }
+
+    public let webAuthnErrorCase: WebAuthnErrorCases
+    public let platformError: Error?
+    public let message: String?
+}
```

Comment:
@george-bafaloukas-forgerock Have you considered keeping the old enum as it was and adding associated values to the cases for platformError and message? For example **case badData(platformError: Error?, message: String?)** .I think that way code would be a bit cleaner and we wouldn't have a struct that extends FRError containing an enum that extends FRError. I tried this approach locally and seems to work just fine.

Reply:
I don't this would work as we have overlapping .notAllowed error cases that might come with different errors. Also some of the rest need extra message information

Reply:
Let me have a closer look on this one

Reply:
i like this approach.